src/components/figures/figure_chart/figure_chart.xml

```diff
@@ -6,7 +6,7 @@
       t-on-contextmenu.prevent.stop="(ev) => !env.isDashboard() and this.onContextMenu(ev)">
       <div class="o-chart-menu" t-if="!env.isDashboard()">
         <div
-          class="o-chart-menu-item"
+          t-att-class="{ 'o-chart-menu-item': true, 'd-none': env.model.getters.isReadonly() }"
```

Comment:
> why not add
> 
> ```
> t-if="!env.model.getters.isReadonly()"
> ```
> 
> instead of hiding the component, which makes it still in the DOM.

If we directly use the `t-if` attribute inside the `o-chart-menu-item` div element then it will affect the behavior of  `o-figure-menu-item o-chart-external-link` which is extended on the position of `o-chart-menu-item` in odoo.

Somewhere here: https://github.com/odoo/odoo/blob/16.0/addons/spreadsheet/static/src/chart/odoo_menu/chart_figure.xml

If we go with this approach, We need to create a PR in the community to extend it on the basis of `o-figure-menu` instead.

What do you suggest ?

Reply:
well... to use `t-if`.
I think you are mislead about how the templating works.  As you mentioned, we extend this template in Odoo but the resolution of the extension is done before the evaluation, meaning that it does not care about the presence of a t-if statement. and the possibility that this `o-chart-menu-item`  will or not be visible at the end. Unless you tried it and the tests failed somehow ? but I doubt it.

Reply:
> well... to use `t-if`. I think you are mislead about how the templating works. As you mentioned, we extend this template in Odoo but the resolution of the extension is done before the evaluation, meaning that it does not care about the presence of a t-if statement. and the possibility that this `o-chart-menu-item` will or not be visible at the end. Unless you tried it and the tests failed somehow ? but I doubt it.

Previously, I was unable to find a way to open a spreadsheet in read-only mode, so I had made an assumption. However, I have now found the correct way and it appears to be functioning correctly.

I had implemented it as suggested by you. Thank you for your guidance on this :)